### ISSUE TITLE
call_group() fails if null is passed as an argument

### ISSUE THREAD

**Issue by JordanHoffman** [godotengine/godot#35833]
Hello, I've barely submitted bugs so sorry if I do this wrong.

**Godot version:**
3.2

**OS/device including version:**
Windows 10
**Issue description:**
If you pass null as one of the arguments for the function that you intend the group being called to execute, then call_group() fails. ex) `get_tree().call_group("theGroup", "theFunction", null)`
The code will continue to run, but the function never actually gets called within the group.

**Steps to reproduce:**
within the intended group member, define your function which can accept an argument (untyped). Now from the script which calls call_group(), just pass null as the argument. It will fail.

**Minimal reproduction project:**
EDIT: added a sample project
[Project.zip](https://github.com/godotengine/godot/files/4143508/Project.zip)

**Comment by Calinou** [comment]
@learis Please upload a minimal reproduction project to make this easier to troubleshoot.

**Comment by learis** [comment]
@Calinou Okay here's an example. You can see via the console log screen that the target group's function is only successfully being called with a non-null argument. The null argument never executes the call.

[Project.zip](https://github.com/godotengine/godot/files/4143505/Project.zip)

**Comment by sumit0190** [comment]
I am not entirely certain if this is the root-cause, but the way variable arguments are parsed is suspect here. For example, in `object.cpp`, we count the number of arguments passed to the function like so:
```cpp
for (int i = 0; i < VARIANT_ARG_MAX; i++) {
    if (argptr[i]->get_type() == Variant::NIL)
        break;
    argc++;
}
```
Which means that if you were to pass a `null` value, it wouldn't be counted at all. I verified this by printing out `argcount` used by `Object::call` - on the first function call, with the `null` passed into the function, `argcount` is `0`, while with a non-null value, `argcount` is `1`.

I have a fix in mind but would need some more help by an experienced hand.

**Comment by learis** [comment]
Just a suggestion. Perhaps you guys can have it check arguments the way signal function handlers check arguments, because they properly recognize null arguments. I have created another sample project, except it uses the signal strategy rather than the group strategy, but does the exact same thing. You can see that it successfully accepts null and non null arguments.
[Signal_Example.zip](https://github.com/godotengine/godot/files/4151549/Signal_Example.zip)

**Comment by ntfshard** [comment]
Wow, this bug is much more interesting that I thought.

TLDR: Old macro `VARIANT_ARG_DECLARE` bring us to a bunch of interesting problems. 

AFAIS `Variant(nul)` is used as \0 for cstrings. `VARIANT_ARG_DECLARE` is defined as `const Variant& p_arg1, ..., const Variant& p_arg5`. `VARIANT_ARGPTRS` is defined as array of 5 pointers to 5 args.
So we have at least 2 problems: can't pass null and you can't use function with more than 5 args! Actually if you have function with 5 args and use `call_group` (see attachment) with more than 5 args rest of them will be ignored without errors
[Project_long_grp_call.zip](https://github.com/godotengine/godot/files/4828128/Project_long_grp_call.zip)

Some functions like `call()` looks like fixed with `Vector<Variant>`(see attachment). I'll try to find solution but without any obligations

**Comment by justinbowes** [comment]
This bit me too. Use case: a group of nodes scattered throughout the tree, listening to select/deselect events. `selected(Node2D)` works fine as a selection broadcast. `selected(null)` fails as a deselection broadcast.

**Comment by Calinou** [comment]
@ntfshard Out of curiosity, did you find a solution to this bug?

**Comment by ntfshard** [comment]
> @ntfshard Out of curiosity, did you find a solution to this bug?

Unfortunately had not enough time. I understand how binding with converting macro->`Vector<Variant>` works but have no clue how to bind `Vector<Variant>` as an argument.

**Comment by Kromey** [comment]
I just ran into this bug yesterday, and took hours of troubleshooting to figure out what was going on. Seems like there should be an error in the editor if calling the function fails.

At the very least can the documentation be updated to note this "`null` issue"? The 5-argument limit is called out, but there's absolutely nothing about `null`s terminating the argument list.

**Comment by DrJamgo** [comment]
Just ran into the issue today.
I am using `false` now instead of null, which reads the same as for checking null in the callback:

```
if myarg:
    # do something with object
else:
    # false was used
```
